### DISCUSSION TITLE
Order of Complexity - Redis versus kvrocks

### DISCUSSION THREAD

**Discussion by adulau** [apache/kvrocks#1686] [General]
As we use more and more kvrocks, we have sometime some fundamental question about the Order of Complexity (O) in kvrocks compared to the ones documented in Redis. As the back-end is significantly different, I assume the O is significantly different per command. Is there a simple way to know the O for the kvrocks command or is this a hidden document somewhere? ;-)

**Comment by mapleFU** [comment]
Redis holds everything in memory, so, though sometime it would has high "cpu cache miss", it can well estimated the complexity by the algorithm.
kvrocks depends on RocksDB and Block Cache in RocksDB. So, it might introduce disk io, which might be spike or sometime bottleneck of the command. And compaction can optimize this.
Besides, I think we could provide some complexity like alibaba tair ( https://help.aliyun.com/zh/tair/developer-reference/doc?spm=a2c4g.11186623.0.0.3d696dc9kUxjBS )
